In [40]:
import pandas as pd
import matplotlib as plt


In [41]:
cities = pd.read_csv('cities.csv')

Commençons par afficher les quelques premières lignes du fichier, pour voir ce que l'on a à traiter :

In [42]:
cities.head()

station_id   city_name      country       state iso2 iso3   latitude  \
0      41515    Asadabad  Afghanistan       Kunar   AF  AFG  34.866000   
1      38954    Fayzabad  Afghanistan  Badakhshan   AF  AFG  37.129761   
2      41560   Jalalabad  Afghanistan   Nangarhar   AF  AFG  34.441527   
3      38947      Kunduz  Afghanistan      Kunduz   AF  AFG  36.727951   
4      38987  Qala i Naw  Afghanistan     Badghis   AF  AFG  34.983000   

   longitude  
0  71.150005  
1  70.579247  
2  70.436103  
3  68.872530  
4  63.133300

Taille du data set et nom des colonnes :

In [43]:
cities.shape, cities.columns

((1245, 8),
 Index(['station_id', 'city_name', 'country', 'state', 'iso2', 'iso3',
        'latitude', 'longitude'],
       dtype='object'))

Nous avons huit colonnes et 1245 villes répertoriées dans ce fichier.\
Remplaçons l'index du tableau par les noms de villes :


In [44]:
cities=cities.set_index('city_name')
cities.head()

station_id      country       state iso2 iso3   latitude  longitude
city_name                                                                     
Asadabad        41515  Afghanistan       Kunar   AF  AFG  34.866000  71.150005
Fayzabad        38954  Afghanistan  Badakhshan   AF  AFG  37.129761  70.579247
Jalalabad       41560  Afghanistan   Nangarhar   AF  AFG  34.441527  70.436103
Kunduz          38947  Afghanistan      Kunduz   AF  AFG  36.727951  68.872530
Qala i Naw      38987  Afghanistan     Badghis   AF  AFG  34.983000  63.133300

Nous avons 1245 villes, regardons à combien de pays différents cela correspond: 

In [45]:
pays = cities['country'].unique()
print(len(pays))


216


L'étude porte donc sur 1245 villes dans 216 pays différents

Regardons les types de données que l'on possède :

In [46]:
cities.dtypes

station_id     object
country        object
state          object
iso2           object
iso3           object
latitude      float64
longitude     float64
dtype: object

Toutes les données sont des objets, sauf la latitude et la longitude, ce qui est logique; nous n'avons pour l'instant aucunement besoin de changer les types puisqu'ils sont cohérents. \
Quantité de données différentes par colonne:

In [47]:
for colonne in cities.columns:
    print(len(cities[colonne].unique()))

1227
216
1150
216
216
1230
1229


Les données sont réparties comme suit :\
1227 station_id (donc certaines villes en ont une similaire)\
216 pays (comme vu précédemment)\
1150 state (certaines villes n'ont pas de state renseigné)\
216 iso2 et 216 iso3 donc avoir 2 ou 3 lettres suffit à différencier ici les 216 pays\
1230 données de latitudes et 1229 données de longitudes; donc certaines villes on exactement la même latitude ou longitude, ce qui n'a rien d'étonnant. \
Voyons la quantité de données manquantes:


In [48]:
print(cities.isnull().sum())

station_id     0
country        0
state         28
iso2           6
iso3           0
latitude       0
longitude      0
dtype: int64


Il manque seulement la donnée de 28 états.

Plus généralement, affichons les données statistiques de chaque colonne:


In [54]:
for colonne in cities.columns:
    print(colonne,':\n',cities[colonne].describe(), '\n')

station_id :
 count      1245
unique     1227
top       78397
freq          2
Name: station_id, dtype: object 

country :
 count       1245
unique       216
top       Russia
freq          77
Name: country, dtype: object 

state :
 count        1217
unique       1149
top       Unknown
freq           32
Name: state, dtype: object 

iso2 :
 count     1239
unique     215
top         RU
freq        77
Name: iso2, dtype: object 

iso3 :
 count     1245
unique     216
top        RUS
freq        77
Name: iso3, dtype: object 

latitude :
 count    1245.000000
mean       23.729983
std        25.996144
min       -54.790003
25%         8.099982
50%        31.611105
75%        43.699980
max        78.216684
Name: latitude, dtype: float64 

longitude :
 count    1245.000000
mean       20.782440
std        66.268905
min      -176.174022
25%        -6.755003
50%        23.237002
75%        63.628302
max       179.216647
Name: longitude, dtype: float64 

